src/lib/libast/tests/string/chresc.c

```diff
@@ -0,0 +1,117 @@
+#include "config_ast.h"  // IWYU pragma: keep
+
+#include <stdint.h>
+#include <string.h>
+
+#include "ast.h"
+#include "terror.h"
+
+struct test {
+    const char *input;
+    int result;
+};
+
+// Test for octal numbers. It also checks if next string is returned correctly.
+void test_octal() {
+    char *input = "\\012\\345\\670";
+    char *nextstr = input;
+    int next, expected;
+
+    next = chresc(input, &nextstr);
+
+    expected = 10;
+    if (next != expected) terror("chresc() :: Failed to convert octal string \\012 :: Expected: %d, Actual: %d", expected, next);
+
+    next = chresc(nextstr, &nextstr);
+    expected = 229;
+    if (next != expected) terror("chresc() :: Failed to convert octal string \\345 :: Expected: %d, Actual: %d", expected, next);
+
+    next = chresc(nextstr, &nextstr);
+    expected = 440;
+    if (next != expected) terror("chresc() :: Failed to convert octal string \\670:: Expected: %d, Actual: %d", expected, next);
+}
+
+// Test for hexadecimal numbers
+void test_hex() {
+    int actual, expected;
+
+    struct test tests[] = {
+        { "\\x01", 1 },
+        { "\\x23", 35 },
+        { "\\x45", 69 },
+        { "\\x67", 103 },
+        { "\\x89", 137 },
+        { "\\xab", 171 },
+        { "\\xcd", 205 },
+        { "\\xef", 239 },
+        { "\\xAB", 171 },
+        { "\\xCD", 205 },
+        { "\\xEF", 239 },
+        { "\\x{f}", 15 },
+        { "\\x[f]", 15 },
+        { NULL, 0 }
+    };
+
+    for (int i=0; tests[i].input; ++i) {
+        actual = chresc(tests[i].input, NULL);
+        expected = tests[i].result;
+        if (actual != expected) terror("chresc() failed :: Failed to convert '%s' :: Expected: %d, Actual: %d", tests[i].input, expected, actual);
+    }
+}
+
+void test_escape_sequences() {
+    int actual, expected;
+    const char *input;
+
+    struct test tests[] = {
+        { "\\a", CC_bel },
+        { "\\b", 8 },
```

Comment:
Here too using symbolic versions rather than decimal constants would be clearer; e.g., `{ "\\b", '\b' }`.